src/shared/components/ButtonBase/ButtonBase.style.ts

```diff
@@ -234,8 +234,8 @@ export const BorderWrapper = styled.div<Pick<ButtonBaseStyleProps, 'textOnly'>>`
   display: flex;
   align-items: center;
   box-sizing: border-box;
-  margin-top: -0.5px;
-  margin-bottom: -0.5px;
+  margin-top: ${({ textOnly }) => textOnly && '-0.5px'};
+  margin-bottom: ${({ textOnly }) => textOnly && '-0.5px'};
```

Comment:
just for clarification, the buttons had some weird heights. Large buttons had 47px, medium 39px and small 31. These lines should fix that, all the sizes should be divisible by 4 now.